src/scripts/circleci-job-canceller/long_job_canceller.py

```diff
@@ -68,7 +85,26 @@ def find_old_workflow_ids(repo_slug, window_start, window_end, headers):
                     yield {"job_status": "too_old", "name": current_workflow['name'], "id": current_workflow['id'], "username": username}
 
 
-def main(circleapitoken, orgreposlug, n_windows, output_file, commit):
+def matches_any_in_list(str, list):
+    for current in list:
+        if ( str.find(current) > -1 ):
+            return True
+    return False
+
+
+def has_only_ignored_jobs(current_info, ignore, headers):
+    pending_approvals = get_workflow_pending_approval_jobs(current_info['id'], headers)
+
+    for current in itertools.filterfalse(
```

Comment:
Any reason to have the `itertools` + `lambda` here? Seems harder/less Pythonic to read than a:

```py
for current in get_workflow_pending_approval_jobs(current_info['id'], headers):
  if not matches_any_in_list(current['name], ignore):
    return False
return True
```